src/org/jitsi/impl/neomedia/rtp/TransportCCEngine.java

```diff
@@ -330,41 +394,53 @@ public void tccReceived(RTCPTCCPacket tccPacket)
             }
         }
 
-        long referenceTime = tccPacket.getReferenceTime();
-        if (lastTimestampUs == -1)
-        {
-            currentOffsetMs = System.currentTimeMillis();
-        }
-        else
+        RTCPTCCPacket.PacketMap packetMap = tccPacket.getPackets();
+        long previousArrivalTimeMs = -1;
+        for (Map.Entry<Integer, Long> entry : packetMap.entrySet())
         {
-            long delta = referenceTime - lastTimestampUs;
-
-            // Detect and compensate for wrap-arounds in base time.
-            if (Math.abs(delta - kBaseTimestampRangeSizeUs) < Math.abs(delta))
-            {
-                delta -= kBaseTimestampRangeSizeUs;  // Wrap backwards.
-            }
-            else if (Math.abs(delta + kBaseTimestampRangeSizeUs) < Math.abs(delta))
+            long arrivalTime250Us = entry.getValue();
+            if (arrivalTime250Us == -1)
             {
-                delta += kBaseTimestampRangeSizeUs;  // Wrap forwards.
+                continue;
             }
 
-            currentOffsetMs += delta / 1000;
-        }
+            if (remoteReferenceTimeMs == -1)
+            {
+                remoteReferenceTimeMs = RTCPTCCPacket.getReferenceTime(
```

Comment:
The field is named "remoteReferenceTimeMs", but if I'm reading RTCPTCCPacket.getReferenceTime correctly, it is assigned a timestamp with the 250us format

Reply:
Oh, I'm wrong, I didn't see the "/4"